hardware/client.go

```diff
@@ -71,24 +75,51 @@ type watcherTinkerbell struct {
 // NewCacherClient returns a new hardware Client, configured to use a provided cacher Client
 // This function is primarily used for testing.
 func NewCacherClient(cc cacher.CacherClient) (Client, error) {
-	var hg Client
-
-	switch dataModelVersion {
-	case "1":
+	if dataModelVersion != "" {
 		return nil, errors.New("NewCacherClient is only valid for the cacher data Model")
-	default:
-		hg = clientCacher{client: cc}
 	}
 
-	return hg, nil
+	return clientCacher{client: cc}, nil
 }
 
 // NewClient returns a new hardware Client, configured appropriately according to the mode (Cacher or Tink) Hegel is running in.
 func NewClient() (Client, error) {
 	var hg Client
 
 	switch dataModelVersion {
-	case "1":
+	case datamodel.Kubernetes:
+		kubeAPI := os.Getenv("HEGEL_KUBE_API")
```

Comment:
I'm trying to standardize components on the same flags. Can we go with flags for `kubernetes-api` and `kubeconfig` and then ingest with viper here?

Reply:
Hegel doesn't support CLI flags at present hence I purposely avoided adding anything for this PR. 

I'd like to address it separately because I think it'd be worth introducing CLI arguments holistically; does that work for you?